File: irrgarten/src/irrgarten/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 * @author Juan Molina Molina
 * @author Paula Jiménez Ruiz
 */
class Monster extends LabyrinthCharacter{
    
    private static int INITIAL_HEALTH=5;
    
    public Monster(String name, float intelligence, float strenght) {
    
        super(name, intelligence, strenght, INITIAL_HEALTH);    // PREGUNTAR POR EL 5
    }
    
    @Override
    public float attack() {
    
        Dice dice=new Dice();
        float attack=dice.intensity(getStrength());
        return attack;
    }
    
    @Override
    public boolean defend(float receivedAttack) {

        boolean isDead=dead();
        if(!isDead) {
        
            Dice dice= new Dice();
            float defensiveEnergy=dice.intensity(getIntelligence());
            
            if(defensiveEnergy<receivedAttack) {
            
                gotWounded();
                isDead=dead();
            }
        }
        
        return isDead;
    }
    
    @Override
    public String toString() { 
    
        return "M"+super.toString();
    }
}

File: ssh/Monster.rb
#encoding:utf-8

require_relative 'LabyrinthCharacter'

module Irrgarten
	
	class Monster < LabyrinthCharacter
		private
			@@INITIAL_HEALTH=5
		public
			def initialize(n,i,s) 
				super(n,i,s,@@INITIAL_HEALTH)
			end
			
			def attack
				return Dice.intensity(get_strength)
			end
			
			def defend(rA)
				is_dead=dead()
				if !is_dead
					defensive_energy=Dice.intensity(get_intelligence)
					if defensive_energy<rA
						got_wounded()
						is_dead=dead()
					end
				end
				
				return is_dead
			end
			
			def to_s
				return "M"+super 
			end
	end
end
			
			
